client/jackson/src/main/kotlin/net/corda/jackson/StringToMethodCallParser.kt

```
@@ -69,15 +69,19 @@ import kotlin.reflect.jvm.kotlinFunction
  *     "addNote id: b6d7e826e8739ab2eb6e077fc4fba9b04fb880bb4cbd09bc618d30234a8827a4, note: Some note"
  */
 @ThreadSafe
-open class StringToMethodCallParser<in T : Any>(targetType: Class<out T>,
-                                                private val om: ObjectMapper = JacksonSupport.createNonRpcMapper(YAMLFactory())) {
+open class StringToMethodCallParser<in T : Any> @JvmOverloads constructor(
+        targetType: Class<out T>,
+        private val om: ObjectMapper = JacksonSupport.createNonRpcMapper(YAMLFactory()))
+{
     /** Same as the regular constructor but takes a Kotlin reflection [KClass] instead of a Java [Class]. */
     constructor(targetType: KClass<out T>) : this(targetType.java)
 
     companion object {
         @Suppress("PLATFORM_CLASS_MAPPED_TO_KOTLIN")
         private val ignoredNames = Object::class.java.methods.map { it.name }
-        private fun methodsFromType(clazz: Class<*>) = clazz.methods.map { it.name to it }.toMap().filterKeys { it !in ignoredNames }
+        private fun methodsFromType(clazz: Class<*>): Map<String, Method> {
+            return clazz.methods.filterNot { it.isSynthetic && it.name !in ignoredNames }.map { it.name to it }.toMap()
```

You're keying on the method name which will cause any subsequent overloaded methods to be silently ignored. Best to throw an exception I think.